src/main/java/parsing/Parser.java

```diff
@@ -314,4 +336,54 @@ private void updateProgressBar(int lineCount, int nol) {
     public static Thread getThread() {
         return parser;
     }
+
+    /**
+     * Reads all genomes of the gfa file and caches them.
+     * @param gw the writer used to write to the cache.
+     * @param line the line on which all genomes are listed.
+     * @return the array of all genomes in the gfa file.
+     */
+    private String[] generateGenomes(BufferedWriter gw, String line) {
+        String str = line.substring(line.indexOf(':') + 1);
+        str = str.substring(str.indexOf(':') + 1);
+        if (str.contains("\t")) {
+            str = str.substring(0, str.indexOf("\t"));
+        }
+        String[] allGenomes = str.split(";");
+        try {
+            gw.write(allGenomes.length + "\t");
+            for (int i = 0; i < allGenomes.length; i++) {
+                gw.write(allGenomes[i] + "\t");
+            }
+            gw.write("\n");
+            gw.flush();
+        } catch (IOException e) {
```

Comment:
I would say, you should make the method throw the IOException since you pass the writer to the method. This is not wrong though.